airflow/serialization/serialized_objects.py

```diff
@@ -970,6 +997,14 @@ def deserialize_dag(cls, encoded_dag: Dict[str, Any]) -> 'SerializedDAG':
             if serializable_task.subdag is not None:
                 setattr(serializable_task.subdag, 'parent_dag', dag)
 
+            if isinstance(task, MappedOperator):
+                for d in (task.mapped_kwargs, task.partial_kwargs):
+                    for k in d:
+                        if not isinstance(d[k], _XcomRef):
+                            continue
+
+                        d[k] = XComArg(operator=dag.get_task(d[k].task_id), key=d[k].key)
```

Comment:
```suggestion
                    for k, v in d.items():
                        if not isinstance(v, _XcomRef):
                            continue

                        d[k] = XComArg(operator=dag.get_task(v.task_id), key=v.key)
```
Wondering if this would be easier to read this way

Reply:
Oh yes, probably. Last night I was thinking we can't change the dict in place, but that's only the case for adding/removing items, but changing existing ones will be fine.